### PR TITLE
RavenDB-15198: merge bulk insert commands when there are a lot of ope…

### PR DESCRIPTION
**Pull request by garayx**
…rations

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs [hunk 1]
```diff
@@ -71,7 +72,7 @@ private async Task<IOperationResult> DoBulkInsert(Action<IOperationProgress> onP
                                     // if we are going to wait on the network, flush immediately
                                     if ((task.Wait(5) == false && numberOfCommands > 0) ||
                                         // but don't batch too much anyway
-                                        totalSize > 16 * Voron.Global.Constants.Size.Megabyte)
+                                        totalSize > 16 * Voron.Global.Constants.Size.Megabyte || operationsCount >= 8192)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs:75]: Do we have some numbers for this? What if this is 4/8/16/32/64/128k ?

File: test/BenchmarkTests/Storing/BulkInsert.cs [hunk 2]
```diff
@@ -37,6 +38,51 @@ public async Task Large_Size_1M()
             await Large_Size_Internal(1_000_000);
         }
 
+        [Fact]
+        public async Task VeryBigTransactionsForBulkInsertOfCounters()
```
Comment (ppekrol) [inline test/BenchmarkTests/Storing/BulkInsert.cs]: Meaningful names here please. `Counters_1M_1Value_1User()` or sth like that.

File: test/BenchmarkTests/Storing/BulkInsert.cs [hunk 3]
```diff
@@ -37,6 +38,51 @@ public async Task Large_Size_1M()
...
+                const string baseDocId = "users/";
+                using (var bulk = store.BulkInsert())
+                {
+                    var docId = baseDocId + 1;
+                    await bulk.StoreAsync(new User() { Name = "Grisha" }, docId);
+
+                    var counters = bulk.CountersFor(docId);
+                    for (var j = 0; j < 1_000_000; j++)
+                    {
+                        var rnd = new Random(DateTime.Now.Millisecond);
```
Comment (ppekrol) [inline test/BenchmarkTests/Storing/BulkInsert.cs]: no randoms. needs to be the same every day.

File: src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs:75]: yes I tested bulk insert of counter, ts, docs, for various values (1k to 20k), 8k gives the fastest result

**Pull request merged**